### ISSUE TITLE
Allow setting content insets for scroll views (like in iOS, example provided)

### ISSUE THREAD

**Issue by Jan-pp** [flutter/flutter#60117]
I tried to address this issue before but I was misunderstood, so let me try again:

In iOS it is possible to set `contentInset` on a `ScrollView`. This basically indicates the unsafe areas that may be obscured. Whenever you activate a text field for example it makes sure it is visible in the safe zone. If you tell it to scroll a certain cell to top it will go to the top of the save zone.

This native example has top and bottom insets of 150. These unsafe areas are colored orange. It behaves as I explained:

![ezgif com-optimize-3](https://user-images.githubusercontent.com/56071132/85439837-b92d6e80-b58d-11ea-9ee4-de0c99483792.gif)

Now for Flutter I don't believe it is possible set insets on a scroll view. Yes you can add padding but these are not considered unsafe areas.

Is the iOS behavior possible in Flutter at all? I can't seem to find it. Thank you.

----------------------

_**While you're at it**:_

In the example you can also see that the scroll indicator has adjusted insets too. Is that possible to do in Flutter? I used `flashScrollIndicators()` to flash it, does Flutter's ScrollBar have a method to flash it? I don't believe it has.

**Comment by iapicca** [comment]
https://github.com/flutter/flutter/issues/59314

**Comment by Piinks** [comment]
Hi @Jan-pp!

> I tried to address this issue before but I was misunderstood, so let me try again:

Is this the same issue as #59314? If so we should close one of them so avoid further confusion.


> Is the iOS behavior possible in Flutter at all? I can't seem to find it. Thank you.

Have you tried providing your own custom insets by creating your own `MediaQueryData` above your scroll view? You can set the `MediaQueryData.viewInsets/viewPadding/padding` to be whatever value you would like then. This could even work to manipulate your own `SafeArea`, as one example, your custom `MediaQuery` > `SafeArea` > `ScrollView`.

> _**While you're at it**:_
> 
> In the example you can also see that the scroll indicator has adjusted insets too. Is that possible to do in Flutter? I used `flashScrollIndicators()` to flash it, does Flutter's ScrollBar have a method to flash it? I don't believe it has.

I would recommend filing a separate issue for this.

**Comment by Jan-pp** [comment]
I can't find a way to change the MediaQuery data, all I found is `MediaQuery.removeViewPadding()`. Please explain how this would work. I also tried `SafeArea` but that just adds bars. I tried lots of combinations to no avail.

Funny thing is that if you use a `Scaffold` with `extendBodyBehindAppBar` enabled and an `AppBar` with a transparent `backgroundColor` and put e.g. a `ListView` in its `body` it behaves the way I'm looking for. So the system is there but I just don't have much control over it.

Yes you can close the other issue, thank you.

**Comment by Piinks** [comment]
You can actually created your own `MediaQueryData` by wrapping that part of your widget tree in a `MediaQuery`, like this

```dart
...
Scaffold(
  body: MediaQuery(
    data: MediaQueryData(
      viewPadding: EdgeInsets.only(bottom: 10.0), // Whatever values you like!
      padding: EdgeInsets.symmetric(top: 100.0),
      viewInsets: EdgeInsets.all(50.0),
    ),
    child: ScrollView(... ),
  ),
),
...
```

I would also see about asking this question on StackOverflow, you may be able to get more specific help on how to use `MediaQuery`.

**Comment by Jan-pp** [comment]
Thanks, that was obvious :p. Problem is when I search for MediaQuery is just find info about getting screen size and things like that.

I tried your example it and doesn't quite work. Changing `textScaleFactor` for example works fine, but `viewPadding` and `viewInsets` does not seem to make any difference, no matter where I put the MediaQuery widget in the tree...

Maybe it's better to post this question on SO, but if you have a working example of MQ in action would be appreciated.

**Comment by goderbauer** [comment]
> Maybe it's better to post this question on SO, but if you have a working example of MQ in action would be appreciated.

SO may be a better place for this.
